anomalib/models/ganomaly/torch_model.py

```diff
@@ -0,0 +1,227 @@
+"""Torch models defining encoder, decoder, Generator and Discriminator."""
+
+# Copyright (C) 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+
+import math
+
+from torch import Tensor, nn
+
+
+class Encoder(nn.Module):
+    """Encoder Network.
+
+    Args:
+        input_size (int): Size of input image
+        latent_vec_size (int): Size of latent vector z
+        num_input_channels (int): Number of input channels in the image
+        n_features (int): Number of features per convolution layer
+        extra_layers (int): Number of extra layers since the network uses only a single encoder layer by default.
+            Defaults to 0.
+    """
+
+    def __init__(
+        self,
+        input_size: int,
+        latent_vec_size: int,
+        num_input_channels: int,
+        n_features: int,
+        extra_layers: int = 0,
+        add_final_conv_layer: bool = True,
+    ):
+        super().__init__()
+
+        assert input_size % 16 == 0, "Input size should be a multiple of 16"  # Why?
+
+        self.model = nn.Sequential()
+
+        self.model.add_module(
+            f"initial-conv-{num_input_channels}-{n_features}",
+            nn.Conv2d(num_input_channels, n_features, kernel_size=4, stride=2, padding=1, bias=False),
+        )
+        self.model.add_module(f"initial-relu-{n_features}", nn.LeakyReLU(0.2, inplace=True))
```

Comment:
I find this module a bit hard to read. Maybe instead of collecting all the layers in the `self.model` attribute, you could separate them into multiple modules, and then call them in the `forward` method? Something like
```
def __init__():
    self.initial_conv = nn.Conv2d(...)
    self.initial_relu = nn.LeakyReLU(...)
    ...

def forward(x):
    x = self.init_conv(x)
    x = self.init_relu(x)
    ...
```
I always like to be able to understand the data flow in a torch module by just looking at the forward mnethod.